mne_bids/utils.py

```diff
@@ -591,12 +591,14 @@ def _find_matching_sidecar(bids_fname, bids_root, suffix, allow_fail=False):
 
     # We only use subject and session as identifier, because all other
     # parameters are potentially not binding for metadata sidecar files
-    search_str = 'sub-' + params['sub']
+    sub_id = 'sub-' + params['sub']
+    search_str = sub_id
     if params['ses'] is not None:
         search_str += '_ses-' + params['ses']
 
-    # Find all potential sidecar files, doing a recursive glob from bids_root
-    search_str = op.join(bids_root, '**', search_str + '*' + suffix)
+    # Find all potential sidecar files, doing a recursive glob
+    # from bids_root/sub_id/
+    search_str = op.join(bids_root, sub_id, '**', search_str + '*' + suffix)
```

Comment:
will you miss candidates that pass the inheritance principle?

Reply:
What do you mean by this?

Reply:
see: https://bids-specification.readthedocs.io/en/stable/02-common-principles.html#the-inheritance-principle. 

Basically, the metadata for `sub_01_task-blah_run-01_meg.fif` and `sub_01_task-blah_run-02_meg.fif` can be stored in a single file.

Reply:
Ah yes it should be, because, the search str is `sub-01_session_blah*meg.json`, with anything in between at least in the code base so far.